### PR TITLE
Namespace as JekyllAdmin

### PR DESCRIPTION
**Pull request by benbalter**
This PR renames the project from `Jekyll/Admin` with a namespace of `Jekyll:Admin` to `Jekyll::JekyllAdmin` which is the Jekyll convention, to allow it to be called as `jekyll-admin` like other `jekyll-*` Gems in the user's config. There should be no substantive changes.

/cc @parkr 

@mertkahyaoglu this will move the `public` folder from `/lib/jekyll/admin` to `/lib/jekyll-admin`. If there are any merge conflicts, you can blow away the public folder on this branch and replace it with your own (but Git should be smart enough to resolve the conflict).

Fixes https://github.com/jekyll/jekyll-admin/issues/25.

### CODE REVIEW

**Comment by benbalter** [comment]
Took a few tries, but got things merged in. @parkr if I can get a LGTM from you, this should be good to go.

File: jekyll-admin.gemspec [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 lib = File.expand_path('../lib', __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-require 'jekyll/admin/version'
+require 'jekyll-admin/version'
 
 Gem::Specification.new do |spec|
   spec.name          = "jekyll-admin"
-  spec.version       = Jekyll::Admin::VERSION
+  spec.version       = Jekyll::JekyllAdmin::VERSION
   spec.authors       = ["Ankur Singh"]
```
Comment (parkr) [inline jekyll-admin.gemspec]: This should be updated.

File: jekyll-admin.gemspec [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
   spec.email         = ["ankur13019@iiitd.ac.in"]
```
Comment (parkr) [inline jekyll-admin.gemspec]: Emails too.

File: lib/jekyll-admin.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+require "jekyll-admin/server/data.rb"
+require "jekyll-admin/server/page.rb"
+require "jekyll-admin/server/static_file.rb"
+
+# Monkey Patche
+require_relative "./jekyll/commands/serve"
+require_relative "./jekyll/convertible_ext"
+
+module Jekyll
+  module JekyllAdmin
```
Comment (parkr) [inline lib/jekyll-admin.rb]: If the file is `lib/jekyll-admin.rb`, convention dictates `Jekyll::Admin`. The namespacing here is `lib/jekyll/jekyll_admin.rb`, which seems to get neither of what you're looking for 😉

**Comment by parkr** [comment]
@benbalter Why not just ship as `JekyllAdmin` proper? No need to include anything in the `Jekyll` namespace unless you're monkey-patching our code 😉

**Comment by benbalter** [comment]
@parkr believe I implemented all your feedback.

@mertkahyaoglu how disruptive would it be if we merged this? In practicality, if you had an existing branch, you'd simply need to merge in master and copy whatever's in the `jekyll/admin/public` folder to `jekyll-admin/public` (blowing away whatever's in there). 

If not too disruptive, I'd actually prefer to merge this sooner than later, as it'll likely cause merge conflicts on the Ruby side with just about any subsequent change.

**Comment by mertkahyaoglu** [comment]
@benbalter No problem for me. We can merge this 👍

**Comment by benbalter** [comment]
@mertkahyaoglu let me know if you have any trouble merging master into the open PRs... as I said, it should be safe to delete the contents of the public folder on master and copy in your content to resolve any conflict... it's just a folder rename.

**Pull request merged**